### PR TITLE
[WB-4456] Allows run deletion from API

### PR DESCRIPTION
**Pull request by annirudh**
https://wandb.atlassian.net/browse/WB-4456

Description
-----------

Allows runs to be deleted from the API.

Testing
-------

- [x] Deleted a run without artifacts
- [x] Deleted a run with artifacts passing in `delete_artifacts=True`
- [x] Unit tests

### CODE REVIEW

**Comment by raubitsj** [comment]
My understanding is this will fail on all local versions except for some of the very latest?
https://github.com/wandb/core/pull/6228
Is that right?
Im ok with that, Just wanted to make sure this was considered.

**Comment by annirudh** [comment]
> My understanding is this will fail on all local versions except for some of the very latest?
> [wandb/core#6228](https://github.com/wandb/core/pull/6228)
> Is that right?
> Im ok with that, Just wanted to make sure this was considered.

Correct, but now that I think about it I believe we can actually make this succeed on much older versions if we only pass along `deleteArtifacts` if it's explicitly set to `True`.

File: wandb/apis/public.py [hunk 1]
```diff
@@ -1024,6 +1024,40 @@ def update(self):
...
+                "$deleteArtifacts: Boolean" if delete_artifacts else "",
+                "deleteArtifacts: $deleteArtifacts" if delete_artifacts else ""
+            )
+        )
+
+        return self.client.execute(
+            mutation,
+            variable_values={
+                "id": self.storage_id,
+                "deleteArtifacts": delete_artifacts,
```
Comment (raubitsj) [inline wandb/apis/public.py:1057]: shouldnt we not pass this deleteArtifacts if delete_artifacts is False?

Comment (annirudh) [reply wandb/apis/public.py:1057]: The extraneous variables are disregarded, so it doesn't cause an issue. I felt it made the code simpler, but I can see it being confusing as well.

Comment (raubitsj) [reply wandb/apis/public.py:1058]: i like cleaner code,  i didnt know that was the behavior.  👍

**Review by raubitsj — APPROVED** [review]

**Pull request merged**